src/main/java/org/apache/sling/commons/scheduler/impl/QuartzThreadPool.java

```diff
@@ -67,8 +72,23 @@ public void setInstanceName(final String name) {
      */
     @Override
     public boolean runInThread(final Runnable job) {
-        this.executor.execute(job);
-
+        synchronized ( this.lock ) {
+            if ( this.counter == 0 ) {
```

Comment:
what about a paranoia check like so?
```
            if ( this.counter <= 0 ) {
```

Reply:
I am not a fan as this is then hiding a bug in the logic and eventually it will cause trouble. Its probably only delaying it